components/post_markdown/post_markdown.tsx

```diff
@@ -1,68 +1,75 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {Post} from 'mattermost-redux/types/posts';
+
 import Markdown from 'components/markdown';
 
-import {renderSystemMessage} from './system_message_helpers.jsx';
+import {TextFormattingOptions, MentionKey} from 'utils/text_formatting';
 
-export default class PostMarkdown extends React.PureComponent {
-    static propTypes = {
+import {renderSystemMessage} from './system_message_helpers';
 
-        /*
+type Props = {
+
+    /*
          * Any extra props that should be passed into the image component
          */
-        imageProps: PropTypes.object,
+    imageProps?: Record<string, any>,
 
-        /*
+    /*
          * Whether or not this text is part of the RHS
          */
-        isRHS: PropTypes.bool,
+    isRHS?: boolean,
 
-        /*
+    /*
          * The post text to be rendered
          */
-        message: PropTypes.string.isRequired,
+    message: string,
 
-        /*
+    /*
          * The optional post for which this message is being rendered
          */
-        post: PropTypes.object,
+    post?: Post & {
```

Comment:
Looks like this is a missing field in the `Post` type. I'll make a fix in `mattermost-redux` so we don't have to do this.

Reply:
https://github.com/mattermost/mattermost-redux/pull/1321